### ISSUE TITLE
Proposal: only use build-templates

### ISSUE THREAD

**Issue by drnic** [cppforlife/knctl#39]
Currently `knctl install` does not install any build templates, and both `knctl deploy` and `knctl build create` do not use them by default (without `--template` flag). Instead, they manually construct build steps that mirror an existing template https://github.com/knative/build-templates/tree/master/kaniko

Perhaps we move towards `knctl install` preinstalling some/all the known build template CRDs, and `knctl deploy` always using one, e.g. `--template kaniko` by default.

**Comment by cppforlife** [comment]
@drnic it's definitely an interesting way. main reason i didnt go that route so far was to avoid figuring out experience of dealing with build templates but may be it's time. also would have to figure out how to inject local source code for -d flag to continue to work.

**Comment by drnic** [comment]
What do you mean "figure out how to inject source code" with build-templates? Isn't this what we can already do now with `--directory=$PWD --template buildpack`?

**Comment by cppforlife** [comment]
> What do you mean "figure out how to inject source code" with build-templates? Isn't this what we can already do now with --directory=$PWD --template buildpack?

yeah you are right, i forgot that that piece doesnt go into build-template.

**Comment by cppforlife** [comment]
pr sounds good. make sure to grab versioned yaml url.

**Comment by drnic** [comment]
Relates to #46
